src/main/java/com/github/junit5docker/DockerExtension.java

```diff
@@ -74,7 +75,11 @@ private void waitForLogAccordingTo(WaitFor waitFor) {
     }
 
     private Supplier<Boolean> findFirstLogContaining(String logToFind) {
-        return () -> dockerClient.logs(containerId).anyMatch(log -> log.contains(logToFind));
+        return () -> {
+            try (Stream<String> logs = dockerClient.logs(containerId)) {
```

Comment:
That's a language feature. I don't think we need a comment. The expected behaviour is covered by a test.